scripts/get-git-tag-name.sh

```diff
@@ -0,0 +1,122 @@
+#!/bin/bash
+
+# This script derives a git tag name from the version fields found in a given Go
+# file. It also checks if the derived git tag name is a valid SemVer compliant
+# version string.
+
+# get_git_tag_name reads the version fields from the given file and then
+# constructs and returns a git tag name.
+get_git_tag_name() {
+  local file_path="$1"
+
+  # Check if the file exists
+  if [ ! -f "$file_path" ]; then
+      echo "Error: File not found at $file_path" >&2
+      exit 1
+  fi
+
+  # Read and parse the version fields.
+  local app_major
+  app_major=$(grep -oP 'AppMajor\s*uint\s*=\s*\K\d+' "$file_path")
+
+  local app_minor
+  app_minor=$(grep -oP 'AppMinor\s*uint\s*=\s*\K\d+' "$file_path")
+
+  local app_patch
+  app_patch=$(grep -oP 'AppPatch\s*uint\s*=\s*\K\d+' "$file_path")
+
+  local app_status
+  app_status=$(grep -oP 'AppStatus\s*=\s*"([a-z]*)"' "$file_path" | cut -d '"' -f 2)
```

Comment:
We could use `grep -oP 'AppStatus\s*=\s*"\K([a-z]*)'` here to avoid needing to use `cut`.

Reply:
Nice, thanks.